vulcan/db/migrations/006_add_workflow_snapshots.rb

```diff
@@ -0,0 +1,29 @@
+Sequel.migration do
+  up do
+    create_table(:workflow_snapshots) do
+      primary_key :id
+
+      foreign_key :figure_id, :figures, null: false, unique: true
+      json :authors, null: false
```

Comment:
Does it possibly make sense to model snapshots as just, a raw yml payload with none of the parsing being normalized?

`cwl_yaml` + `image`?
Also, does image itself need to be stored here?  What if the image url is simply deterministic based on the snapshot id itself?

Reply:
Maybe I'm wrong though, I'm just curious how you felt: what were your thoughts into the table being broken out this way?

Reply:
I wasn't sure if we wanted to keep it as CWL YAML, since at one point we had talked about perhaps moving workflows to a database, etc., and then generating the YAML on the fly. Since the UI-side uses this JSON representation, I thought it would be safe to mirror that, though you're right, it introduces complexity in the conversion. 

I also wasn't sure if this would more closely reflect the format of workflow editing, though since that isn't really well defined, perhaps it shouldn't / doesn't carry much weight at this point, as a consideration.

Reply:
Which metadata again?  Sorry.